File: src/test/java/cernx/colt/matrix/tfloat/AllFloatMatrixTests.java
package cernx.colt.matrix.tfloat;

import cernx.colt.matrix.tfloat.impl.DenseColumnFloatMatrix2DTest;
import cernx.colt.matrix.tfloat.impl.DenseColumnFloatMatrix2DViewTest;
import cernx.colt.matrix.tfloat.impl.DenseFloatMatrix1DViewTest;
import cernx.colt.matrix.tfloat.impl.DenseFloatMatrix2DTest;
import cernx.colt.matrix.tfloat.impl.DenseFloatMatrix2DViewTest;
import cernx.colt.matrix.tfloat.impl.DenseLargeFloatMatrix2DTest;
import cernx.colt.matrix.tfloat.impl.DenseLargeFloatMatrix2DViewTest;
import cernx.colt.matrix.tfloat.impl.DiagonalFloatMatrix2DTest;
import cernx.colt.matrix.tfloat.impl.SparseCCMFloatMatrix2DViewTest;
import cernx.colt.matrix.tfloat.impl.SparseFloatMatrix1DTest;
import cernx.colt.matrix.tfloat.impl.SparseFloatMatrix1DViewTest;
import cernx.colt.matrix.tfloat.impl.SparseFloatMatrix2DViewTest;
import cernx.colt.matrix.tfloat.impl.SparseRCFloatMatrix2DTest;
import cernx.colt.matrix.tfloat.impl.SparseRCFloatMatrix2DViewTest;
import cernx.colt.matrix.tfloat.impl.SparseRCMFloatMatrix2DTest;
import cernx.colt.matrix.tfloat.impl.SparseRCMFloatMatrix2DViewTest;
import junit.framework.Test;
import junit.framework.TestSuite;
import cern.colt.matrix.tfloat.algo.solver.AllFloatMatrixSolverTests;
import cernx.colt.matrix.tfloat.impl.DenseFloatMatrix1DTest;
import cern.colt.matrix.tfloat.impl.DenseFloatMatrix3DTest;
import cern.colt.matrix.tfloat.impl.DenseFloatMatrix3DViewTest;
import cern.colt.matrix.tfloat.impl.DenseLargeFloatMatrix3DTest;
import cern.colt.matrix.tfloat.impl.DenseLargeFloatMatrix3DViewTest;
import cernx.colt.matrix.tfloat.impl.DiagonalFloatMatrix2DViewTest;
import cernx.colt.matrix.tfloat.impl.SparseCCFloatMatrix2DTest;
import cernx.colt.matrix.tfloat.impl.SparseCCFloatMatrix2DViewTest;
import cernx.colt.matrix.tfloat.impl.SparseCCMFloatMatrix2DTest;
import cernx.colt.matrix.tfloat.impl.SparseFloatMatrix2DTest;
import cern.colt.matrix.tfloat.impl.SparseFloatMatrix3DTest;
import cern.colt.matrix.tfloat.impl.SparseFloatMatrix3DViewTest;

public class AllFloatMatrixTests {

    public static Test suite() {
        TestSuite suite = new TestSuite("cern.colt.matrix.tfloat tests");
        suite.addTestSuite(DenseFloatMatrix1DTest.class);
        suite.addTestSuite(DenseFloatMatrix1DViewTest.class);
        suite.addTestSuite(SparseFloatMatrix1DTest.class);
        suite.addTestSuite(SparseFloatMatrix1DViewTest.class);

        suite.addTestSuite(DenseFloatMatrix2DTest.class);
        suite.addTestSuite(DenseFloatMatrix2DViewTest.class);
        suite.addTestSuite(DenseColumnFloatMatrix2DTest.class);
        suite.addTestSuite(DenseColumnFloatMatrix2DViewTest.class);
        suite.addTestSuite(DenseLargeFloatMatrix2DTest.class);
        suite.addTestSuite(DenseLargeFloatMatrix2DViewTest.class);

        suite.addTestSuite(SparseFloatMatrix2DTest.class);
        suite.addTestSuite(SparseFloatMatrix2DViewTest.class);
        suite.addTestSuite(DiagonalFloatMatrix2DTest.class);
        suite.addTestSuite(DiagonalFloatMatrix2DViewTest.class);

        suite.addTestSuite(SparseRCFloatMatrix2DTest.class);
        suite.addTestSuite(SparseRCFloatMatrix2DViewTest.class);
        suite.addTestSuite(SparseRCMFloatMatrix2DTest.class);
        suite.addTestSuite(SparseRCMFloatMatrix2DViewTest.class);

        suite.addTestSuite(SparseCCFloatMatrix2DTest.class);
        suite.addTestSuite(SparseCCFloatMatrix2DViewTest.class);
        suite.addTestSuite(SparseCCMFloatMatrix2DTest.class);
        suite.addTestSuite(SparseCCMFloatMatrix2DViewTest.class);

        return suite;
    }
}

File: src/test/scala/cern/colt/matrix/tfloat/AllFloatMatrixTests.scala
package cern.colt.matrix.tfloat

import junit.framework.{TestCase, Test, TestSuite}
import cern.colt.matrix.tfloat.impl.DenseColumnFloatMatrix2DTest
import cern.colt.matrix.tfloat.impl.DenseColumnFloatMatrix2DViewTest
import cern.colt.matrix.tfloat.impl.DenseFloatMatrix1DTest
import cern.colt.matrix.tfloat.impl.DenseFloatMatrix1DViewTest
import cern.colt.matrix.tfloat.impl.DenseFloatMatrix2DTest
import cern.colt.matrix.tfloat.impl.DenseFloatMatrix2DViewTest
import cern.colt.matrix.tfloat.impl.DenseLargeFloatMatrix2DTest
import cern.colt.matrix.tfloat.impl.DenseLargeFloatMatrix2DViewTest
import cern.colt.matrix.tfloat.impl.DiagonalFloatMatrix2DTest
import cern.colt.matrix.tfloat.impl.DiagonalFloatMatrix2DViewTest
import cern.colt.matrix.tfloat.impl.SparseCCFloatMatrix2DTest
import cern.colt.matrix.tfloat.impl.SparseCCFloatMatrix2DViewTest
import cern.colt.matrix.tfloat.impl.SparseCCMFloatMatrix2DTest
import cern.colt.matrix.tfloat.impl.SparseCCMFloatMatrix2DViewTest
import cern.colt.matrix.tfloat.impl.SparseFloatMatrix1DTest
import cern.colt.matrix.tfloat.impl.SparseFloatMatrix1DViewTest
import cern.colt.matrix.tfloat.impl.SparseFloatMatrix2DTest
import cern.colt.matrix.tfloat.impl.SparseFloatMatrix2DViewTest
import cern.colt.matrix.tfloat.impl.SparseRCFloatMatrix2DTest
import cern.colt.matrix.tfloat.impl.SparseRCFloatMatrix2DViewTest
import cern.colt.matrix.tfloat.impl.SparseRCMFloatMatrix2DTest
import cern.colt.matrix.tfloat.impl.SparseRCMFloatMatrix2DViewTest

object AllFloatMatrixTests extends TestCase {

  def suite(): Test = {
    val suite = new TestSuite("cern.colt.matrix.tfloat tests")
    suite.addTestSuite(classOf[DenseFloatMatrix1DTest])
    suite.addTestSuite(classOf[DenseFloatMatrix1DViewTest])
    suite.addTestSuite(classOf[SparseFloatMatrix1DTest])
    suite.addTestSuite(classOf[SparseFloatMatrix1DViewTest])
    suite.addTestSuite(classOf[DenseFloatMatrix2DTest])
    suite.addTestSuite(classOf[DenseFloatMatrix2DViewTest])
    suite.addTestSuite(classOf[DenseColumnFloatMatrix2DTest])
    suite.addTestSuite(classOf[DenseColumnFloatMatrix2DViewTest])
    suite.addTestSuite(classOf[DenseLargeFloatMatrix2DTest])
    suite.addTestSuite(classOf[DenseLargeFloatMatrix2DViewTest])
    suite.addTestSuite(classOf[SparseFloatMatrix2DTest])
    suite.addTestSuite(classOf[SparseFloatMatrix2DViewTest])
    suite.addTestSuite(classOf[DiagonalFloatMatrix2DTest])
    suite.addTestSuite(classOf[DiagonalFloatMatrix2DViewTest])
    suite.addTestSuite(classOf[SparseRCFloatMatrix2DTest])
    suite.addTestSuite(classOf[SparseRCFloatMatrix2DViewTest])
    suite.addTestSuite(classOf[SparseRCMFloatMatrix2DTest])
    suite.addTestSuite(classOf[SparseRCMFloatMatrix2DViewTest])
    suite.addTestSuite(classOf[SparseCCFloatMatrix2DTest])
    suite.addTestSuite(classOf[SparseCCFloatMatrix2DViewTest])
    suite.addTestSuite(classOf[SparseCCMFloatMatrix2DTest])
    suite.addTestSuite(classOf[SparseCCMFloatMatrix2DViewTest])
//    suite.addTestSuite(classOf[DenseFloatMatrix3DTest])
//    suite.addTestSuite(classOf[DenseFloatMatrix3DViewTest])
//    suite.addTestSuite(classOf[SparseFloatMatrix3DTest])
//    suite.addTestSuite(classOf[SparseFloatMatrix3DViewTest])
//    suite.addTestSuite(classOf[DenseLargeFloatMatrix3DTest])
//    suite.addTestSuite(classOf[DenseLargeFloatMatrix3DViewTest])
    suite
  }
}
